sunpy/map/tests/test_mapbase.py

```diff
@@ -226,7 +226,17 @@ def test_remove_observers(aia171_test_map):
                       match='Missing metadata for observer: assuming Earth-based observer.*'):
         aia171_test_map.observer_coordinate
 
-
+def test_partially_missing_observers(aia171_test_map):
+    del aia171_test_map.meta['dsun_obs']
+    del aia171_test_map.meta['haex_obs']
+    with pytest.warns(SunpyUserWarning,
+                      match="Missing metadata for observer: assuming Earth-based observer.\n" +
+                            "For frame '<class 'astropy.coordinates.builtin_frames.ecliptic.HeliocentricMeanEcliptic'>' " +
+                            "the following metadata is missing :haex_obs\n" +
```

Comment:
Ugh, having the class repr looks ugly!  Two options:

- If `frame` is a class rather than a string, access the attribute `frame.name` here: https://github.com/sunpy/sunpy/blob/d5ee8a910e76349c4de3632abbdebe5cb4b3f38d/sunpy/map/mapbase.py#L642
- Alternatively, we could change the AIA-specific code to use a string to specify the class rather than the class itself (i.e., use `"heliocentricmeanecliptic"` instead of `HeliocentricMeanEcliptic` here: https://github.com/sunpy/sunpy/blob/c78989774567055618ba23533f8927a355e98788/sunpy/map/sources/sdo.py#L66)
It would need a bit of additional contortion until we bump up our Astropy dependency because `HeliocentricMeanEcliptic` doesn't exist in Astropy <3.2.

Reply:
I like the first one the most. You could do that logic higher up where you create `missing_meta` rather than in the f-string.

Reply:
Done
`For frame 'heliocentricmeanecliptic' the following metadata is missing: haex_obs` This will be the new warning message provided 'haex_obs' is missing.
I also chaged the frame in the meta, I believe frame.name would give the user a better idea than the class repr